include/poketch_data.h

```diff
@@ -1,51 +1,98 @@
-#define POKETCH_APPID_COUNTER               10
-#define POKETCH_APPID_ANALOGWATCH           11
-#define POKETCH_APPID_MARKINGMAP            12
-#define POKETCH_APPID_LINKSEARCHER          13
-#define POKETCH_APPID_COINTOSS              14
-#define POKETCH_APPID_MOVETESTER            15
-#define POKETCH_APPID_CALENDAR              16
-#define POKETCH_APPID_DOTART                17
-#define POKETCH_APPID_ROULETTE              18
-#define POKETCH_APPID_RADARCHAINCOUNTER     19
-#define POKETCH_APPID_KITCHENTIMER          20
-#define POKETCH_APPID_COLORCHANGER          21
-#define POKETCH_APPID_MATCHUPCHECKER        22
-#define POKETCH_APPID_STOPWATCH             23
-#define POKETCH_APPID_ALARMCLOCK            24
-#define POKETCH_APPID_MAX                   25
-
-/* Max number of Map Markers. */
-#define POKETCH_MAPMARKER_COUNT             6
+typedef enum PoketchScreenColor {
+    POKETCH_SCREEN_COLOR_GREEN,
+    POKETCH_SCREEN_COLOR_YELLOW,
+    POKETCH_SCREEN_COLOR_ORANGE,
+    POKETCH_SCREEN_COLOR_RED,
+    POKETCH_SCREEN_COLOR_PURPLE,
+    POKETCH_SCREEN_COLOR_BLUE,
+    POKETCH_SCREEN_COLOR_TEAL,
+    POKETCH_SCREEN_COLOR_WHITE,
+
+    POKETCH_SCREEN_COLOR_MAX,
+} PoketchScreenColor;
+
+enum {
+    POKETCH_MAPMARKER_COUNT = 6,
+    POKETCH_REGISTRY_SIZE = 32
+};
+
+/**
+ * @brief All of the Poketch's internally tracked data including settings, registered apps, and the data for some apps (eg. pedometer, alarm clock).
+ */
+typedef struct PoketchData {
+    u8 unk_00_0 : 1;
+    u8 pedometerEnabled : 1;    //!< Whether or not the pedometer is registered and will take step count updates.
+    u8 unk_00_2 : 1;
+    u8 screenColor : 3;         //!< Screen palette color (see PoketchScreenColor enum above)
+    u8 unk_00_6 : 2;            //!< unused; bitfield padding
+
+    s8 appCount;                                //!< Number of currently registered apps
+    s8 appIndex;                                //!< Currently selected app
+    u8 appRegistry[POKETCH_REGISTRY_SIZE];      //!< Registration status of all apps. Indices 0-24 correspond to the App IDs in poketch_data.h. Indices 25-31 are unused.
+    
+    u32 pedometer;         //!< Step counter
+
+    u16 alarmSet : 1;      //!< Whether or not the alarm is currently enabled.
+    u16 alarmHour : 5;     //!< Current Hour setting on the alarm clock.
+    u16 alarmMinute : 6;   //!< Current Minute setting on the alarm clock.
+    u16 unk_28_12 : 4;     //!< unused; bitfield padding
+
+    u8 unk_2A[120];
+    u32 unk_A4;
+    u8 unk_A8;
+
+    /**
+    * @brief XY coordinates of a single Map Marker in the Marking Map app
+    */
+    struct {
+        u8 x;                                       //!< X coordinate of map marker
+        u8 y;                                       //!< Y coordinate of map marker
+    } markMapPositions[POKETCH_MAPMARKER_COUNT];    //!< Map markers
+
+    struct {
+        u16 unk_00;
+        u16 unk_02;
+        u32 unk_04;
+    } unk_B8[12];
+} PoketchData;
 
 /*
```

Comment:
Should these be Doxygen-comments? They're expressed with a double-star on the first line, Javadoc style:

```c
/**
 * [brief]
 *
 * @param
 * @return
 */
```